### PR TITLE
Use Auth0 as a default auth service for Rails projects

### PR DESCRIPTION
**Pull request by tahb**
- should we talk about this change for all projects, rather than just Rails?

### CODE REVIEW

**Comment by pezholio** [comment]
I'm kinda conflicted on this. While I'm a big fan of making our lives easier, I'm a wee bit hesitant on relying on a third party for authentication. Happy to be convinced otherwise though

File: rfc-021-use-auth0-as-the-default-auth-provider.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+We MAY use it to create and manage users rather than building support within each service.
+
+Auth0 offers competitive pricing and forgiving quota allowances.
+
+Auth0 offers an ability to run in proxy mode, this allows forwarding to popular SSO providers such as Okta or Google. This provides us an efficient way to migrate later.
+
+## Next steps
+
+As the majority of clients have auth services it does not seem appropriate to add anything to the rails-template.
```
Comment (erbridge) [inline rfc-021-use-auth0-as-the-default-auth-provider.md]: Where would we document this decision other than here? Playbook?

Comment (tahb) [reply rfc-021-use-auth0-as-the-default-auth-provider.md]: I'm not sure it does, do you think there is a relevant section to the Playbook?

Comment (erbridge) [reply rfc-021-use-auth0-as-the-default-auth-provider.md]: Maybe a guide on how we handle auth in Rails?

Comment (Floppy) [reply rfc-021-use-auth0-as-the-default-auth-provider.md]: I think duplicating it is probably undesirable, but we might want to start keeping lists of read-these-before-you-start ADRs in the `rails-template` README?

Comment (tahb) [reply rfc-021-use-auth0-as-the-default-auth-provider.md]: I don't want to commit to creating a guide in a next step that I can't commit to doing myself. If someone else wants to take that investigation and documentation on then I'd be happy to include it.

Keeping a list of relevant ADRs in the template's README would be helpful in linking this all together when we make a decision but it's not common enough to include default config.

Comment (tahb) [reply rfc-021-use-auth0-as-the-default-auth-provider.md]: I've started the list. When both that and this PR are merged we can add a new item to the list https://github.com/dxw/rails-template/pull/58

Comment (erbridge) [reply rfc-021-use-auth0-as-the-default-auth-provider.md]: > I think duplicating it is probably undesirable

When we discussed using RFCs to make decisions, we wanted to make sure we weren't building an ever growing list of documents to read for new people joining the team. We'll keep the RFCs around even if they're superceded by later decisions, so we can look back on them and understand how we made the decisions we made. So we said that the RFC would never be the only place documenting the **current status** of our practices as a tech team. So output of an RFC should be some documentation or change somewhere else.

At least that was my understanding.

**Pull request merged**